tools/topology/topology2/include/components/micsel.conf

```diff
@@ -0,0 +1,101 @@
+#
+# Common widget micsel
+#
+# A generic micsel widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.micsel.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.micsel."N" {
+#
+#	}
+#
+# Where N is the unique instance number for the micsel object within the same alsaconf node.
+
+Class.Widget."micsel" {
+	#
+	# Pipeline ID for the micsel object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# Micsel object instance
+	#
+	DefineAttribute."instance" {}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	#
+	# micsel component UUID
+	#
+	DefineAttribute."uuid" {
+		type "string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."process_type" {
+		type "string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_process.string"
+	}
+
+	DefineAttribute."payload_with_output_fmt" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.bool"
```

Comment:
doesnt this contradict the new kernel changes proposed by @aiChaoSONG

Reply:
I will submit a separate PR for it after @aiChaoSONG 's PR is merged.

Reply:
@libinyang lets just fix this in one go please. Why make a change that we know won't work and then fix it?